tests/integration/test_integratord/test_integratord_read_json_file_deleted.py

```diff
@@ -0,0 +1,128 @@
+                                            configure_local_internal_options_module, restart_wazuh_daemon_function,
+                                            wait_for_start_module):
+    '''
+    description: Check that if while integratord is reading from the alerts.json file, it is deleted, the expected
+    error message is displayed, and if the file is created again and alerts are inserted, integratord continues
+    working and alerts are read
+    wazuh_min_version: 4.3.5
+    tier: 1
+    parameters:
+        - configuration:
+            type: dict
+            brief: Configuration loaded from `configuration_template`.
+        - metadata:
+            type: dict
+            brief: Test case metadata.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Set wazuh configuration.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Truncate all the log files and json alerts files before and after the test execution.
+        - configure_local_internal_options_module:
+            type: fixture
+            brief: Configure the local internal options file.
+        - restart_wazuh_daemon_function:
+            type: fixture
+            brief: Restart wazuh daemon before starting a test.
+        - wait_for_start_module:
+            type: fixture
+            brief: Detect the start of the Integratord module in the ossec.log
+    assertions:
+        - Verify the expected response with for a given alert is recieved
+    input_description:
+        - The `config_integratord_read_json_alerts.yaml` file provides the module configuration for this test.
+        - The `cases_integratord_read_json_file_deleted` file provides the test cases.
+    expected_output:
+        - r'.*wazuh-integratord.*ERROR.*Could not retrieve information of file.*alerts.json.*No such file.*'
+        - r'.*wazuh-integratord.*alert_id.*\"integration\": \"virustotal\".*'
+
+    '''
+    wazuh_monitor = FileMonitor(LOG_FILE_PATH)
+    command = f"touch {ALERT_FILE_PATH} && chmod 640 {ALERT_FILE_PATH} && chown wazuh:wazuh {ALERT_FILE_PATH}"
+
+    remove_file(ALERT_FILE_PATH)
+    check_integratord_event(file_monitor=wazuh_monitor, timeout=global_parameters.default_timeout*2,
+                            callback=generate_monitoring_callback(integrator.CB_CANNOT_RETRIEVE_JSON_FILE),
+                            error_message=integrator.ERR_MSG_CANNOT_RETRIEVE_MSG_NOT_FOUND)
+    # Create file and insert alert. Wait one second so Integrator detects the file before the insertion
+    run_local_command_returning_output(command)
+    time.sleep(2)
+    run_local_command_returning_output(f"echo '{metadata['alert_sample']}' >> {ALERT_FILE_PATH}")
+
+    # Read Response in ossec.log
+    check_integratord_event(file_monitor=wazuh_monitor, timeout=global_parameters.default_timeout*2,
+                            callback=generate_monitoring_callback(integrator.CB_VIRUSTOTAL_ALERT),
```

Comment:
Add comment

Reply:
done in https://github.com/wazuh/wazuh-qa/pull/3163/commits/6b7c09d4157c76bd09758773696f5ececa5bf18a